pkg/combustion/registry_test.go

```diff
@@ -0,0 +1,155 @@
+package combustion
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+)
+
+func TestWriteHaulerManifestValidManifest(t *testing.T) {
+	// Setup
+	ctx, teardown := setupContext(t)
+	defer teardown()
+
+	ctx.ImageDefinition = &image.Definition{
+		EmbeddedArtifactRegistry: image.EmbeddedArtifactRegistry{
+			ContainerImages: []image.ContainerImage{
+				{
+					Name: "hello-world:latest",
+				},
+				{
+					Name:           "rgcrprod.azurecr.us/longhornio/longhorn-ui:v1.5.1",
+					SupplyChainKey: "carbide-key.pub",
+				},
+			},
+			HelmCharts: []image.HelmChart{
+				{
+					Name:    "rancher",
+					RepoURL: "https://releases.rancher.com/server-charts/stable",
+					Version: "2.8.0",
+				},
+			},
+		},
+	}
+
+	// Test
+	err := writeHaulerManifest(ctx)
+
+	// Verify
+	require.NoError(t, err)
+
+	manifestFileName := filepath.Join(ctx.BuildDir, haulerManifestYamlName)
+	_, err = os.Stat(manifestFileName)
+	require.NoError(t, err)
+
+	foundBytes, err := os.ReadFile(manifestFileName)
+	require.NoError(t, err)
+	found := string(foundBytes)
+	assert.Contains(t, found, "rgcrprod.azurecr.us/longhornio/longhorn-ui:v1.5.1")
+	assert.Contains(t, found, "https://releases.rancher.com/server-charts/stable")
+}
+
+func TestWriteHaulerManifestEmptyManifest(t *testing.T) {
```

Comment:
Is this even valid? Have we tested what happens when the file doesn't contain any images and / or charts?

```
apiVersion: content.hauler.cattle.io/v1alpha1
kind: Images
metadata:
  name: Embedded-Registry-Images
spec:
  images:
---
apiVersion: content.hauler.cattle.io/v1alpha1
kind: Charts
metadata:
  name: Embedded-Registry-Charts
spec:
  charts:
```

Reply:
I was doing this just to show how the function would behave with an empty input, but in reality, it should never actually reach this step in that state. Should I remove the test?

Reply:
Just tested: if either value is empty, it is just skipped, so I'll remove the test.